ci/ci/ci.py

```diff
@@ -395,6 +396,17 @@ async def batch_callback_handler(request):
             for wb in watched_branches:
                 if wb.branch.short_str() == target_branch:
                     log.info(f'watched_branch {wb.branch.short_str()} notify batch changed')
+
+                    if 'test' in attrs and params['complete']:
+                        assert 'deploy' not in attrs
+                        assert 'dev' not in attrs
+                        namespace = json.loads(attrs['namespace'])
+                        assert namespace != 'default'
```

Comment:
Can we assert when this callback gets executed? Mainly I want to make sure this doesn't run in the wrong scope (ie. dev or test).

Reply:
I think it's valid for this callback to run in dev (dev deploy against a dev CI), but I'll add a guard to only change the database in default